### PR TITLE
add SparseTensor.cast & SparseTensor.applyFun

### PR DESCRIPTION
**Pull request by sperlingxx**
## What changes were proposed in this pull request?

implement SparseTensor.cast with its dependency SparseTensor.applyFun

## How was this patch tested?

unit test

### CODE REVIEW

**Comment by yiheng** [comment]
@qiuxin2012 Please take a look at it.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/SparseTensor.scala [hunk 1]
```diff
@@ -1000,14 +1000,24 @@ private[tensor] class SparseTensor[@specialized(Float, Double) T: ClassTag](
     throw new UnsupportedOperationException(s"SparseTensor: Unimplemented method")
   }
 
-  override def applyFun[A : ClassTag](t: Tensor[A], func: (A) => T): Tensor[T] = {
-    throw new UnsupportedOperationException(s"SparseTensor: Unimplemented method")
+  override def applyFun[A: ClassTag](
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/SparseTensor.scala]: add  unit test

**Comment by sperlingxx** [comment]
@qiuxin2012 done

**Comment by qiuxin2012** [comment]
jenkins passed

**Pull request merged**